agent/lib/kontena/websocket_client.rb

```diff
@@ -101,26 +100,31 @@ def send_message(msg)
       error "failed to send message: #{exc.message}"
     end
 
+    # @param [String] method
+    # @param [Array] params
+    def send_notification(method, params)
+      data = MessagePack.dump([2, method, params]).bytes
+      send_message(data)
+    rescue => exc
+      error "failed to send notification: #{exc.message}"
+    end
+
     # @param [Faye::WebSocket::API::Event] event
     def on_open(event)
       ping_timer.cancel if ping_timer
       info 'connection established'
       @connected = true
       @connecting = false
+      notify_actors('websocket:open', event)
```

Comment:
I guess this isn't used anywhere yet?

Not exactly the subject of this PR, but looking at the workers subscribing to `websocket:connect`, I think it might make more sense to have to subscribe to this `websocket:open`?

Mostly it would avoid draining stuff from the agent queue into a websocket that fails to open because the master is down... those messages will get discarded?